pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/realtimetoofflinesegments/RealtimeToOfflineSegmentsTaskGenerator.java

```diff
@@ -317,6 +325,13 @@ private long getWatermarkMs(String realtimeTableName, List<SegmentZKMetadata> co
       realtimeToOfflineSegmentsTaskMetadata = new RealtimeToOfflineSegmentsTaskMetadata(realtimeTableName, watermarkMs);
       _clusterInfoAccessor.setRealtimeToOfflineSegmentsTaskMetadata(realtimeToOfflineSegmentsTaskMetadata);
     }
+
+    _realtimeToOfflineSegmentsTaskMetadata = realtimeToOfflineSegmentsTaskMetadata;
+
     return realtimeToOfflineSegmentsTaskMetadata.getWatermarkMs();
   }
+
+  private void setWatermarkMs(long watermarkMs) {
+    _realtimeToOfflineSegmentsTaskMetadata.setWatermarkMs(watermarkMs);
```

Comment:
this will only set it into the RealtimeToOfflineSegmentsTaskMetadata object. It will not get persisted into zk for the next task.